Assets/HoloToolkit/UX/Scripts/Dialog/Dialog.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity.Receivers;
+using System;
+using System.Collections;
+using UnityEngine;
+
+namespace HoloToolkit.UX.Dialog
+{
+    /// <summary>
+    /// Used to tell simple dialogs which buttons to create
+    /// And to tell whatever launched the dialog which button was pressed
+    /// Can be extended to include more information for dialog construction
+    /// (eg detailed messages, button names, colors etc)
+    /// </summary>
+    public abstract class Dialog : InteractionReceiver
+    {
+        public enum StateEnum
```

Comment:
These enums should be moved outside of the class. Also, please rename them to imply usage (ex: StateEnum ->DialogState and ButtonTypeEnum -> DialogButtonTypes)